### PR TITLE
Use logging instead of print statements

### PR DESCRIPTION
**Pull request by arokem**
Also removing "from future" imports along the way.

### CODE REVIEW

**Comment by skoudoro** [comment]
Overall this PR looks good to me. I see a WIP on your title, are there other files that you want to update?

File: dipy/data/fetcher.py [hunk 1]
```diff
@@ -39,9 +39,9 @@ class FetcherError(Exception):
 
 def _log(msg):
```
Comment (skoudoro) [inline dipy/data/fetcher.py]: Do we really want to keep this function? Since you are using `logging` module, you should delete it.

Comment (arokem) [reply dipy/data/fetcher.py]: It's useful as shorthand. I just pushed a clarification of the docstring.

**Comment by arokem** [comment]
Now also rebased.

**Comment by arokem** [comment]
I'm also taking the opportunity to remove `verbose` settings, because I believe this should be set by the logging level. Would this break something else?

**Comment by skoudoro** [comment]
I think we should keep `verbose`. We just need to improve it and do like the registration module.  Look at `dipy.align.imwarp`

**Comment by skoudoro** [comment]
Hi @arokem, What is your status on this PR? it looks close to being done!

**Comment by arokem** [comment]
Just rebased. IIUC, the main thing that needs to be done is reinstating `verbose` setting?

**Comment by skoudoro** [comment]
Yes, I think so

**Comment by arokem** [comment]
OK. I believe that I have restored the `verbose` flag everywhere we previously had it. I still think that this is something that is best handled by having the users set their logging levels, but we can also do that later if y'all end up agreeing with me.

**Comment by arokem** [comment]
I believe that this ready for a re-review. Codecov is just being dumb.

**Comment by arokem** [comment]
@arokem will look into nicer printing options for logging

**Comment by skoudoro** [comment]
Is there anything you want to add? I forgot what we said during the last online meeting concerning this PR but it looks ready to be merged for me.  Maybe, the logging "_design/string format_" can be done on a new PR. What do you think?

**Comment by arokem** [comment]
Ready on my end. @Garyfallidis : have you had a chance to think about this more?

**Comment by skoudoro** [comment]
can you rebase this one or resolve the conflict @arokem?

**Comment by arokem** [comment]
Rebased!

**Comment by skoudoro** [comment]
Thank you @arokem! Any other comments? will merge it by the end of the week

**Comment by Garyfallidis** [comment]
Hello @arokem. Sorry can you rebase this one more time? Will look into merging the following days.

**Comment by arokem** [comment]
All set.

**Pull request merged**